src/overlays/actors/ovl_Obj_Driftice/z_obj_driftice.c

```diff
@@ -76,200 +87,257 @@ static InitChainEntry sInitChain[] = {
-    *arg3 = arg0->unk_00;
-    Math_ScaledStepToS(&arg0->unk_04, phi_s0, arg0->unk_06);
 
+    Math_ScaledStepToS(&dynamics->precessionAngle, this->dyna.actor.yawTowardsPlayer,
+                       dynamics->precessionAngleRateOfChange);
+    *precessionAngle = dynamics->precessionAngle;
+
+    Math_ScaledStepToS(&dynamics->tiltMagnitude2, targetTiltMagnitude2, dynamics->tiltMagnitude2RateOfChange);
     for (i = 0; i < 2; i++) {
-        temp_s0 = &arg0->unk_0C[i];
+        oscillation = &dynamics->oscillationState[i];
 
-        if (temp_s0->unk_00 > 0) {
-            temp_s0->unk_00--;
+        if (oscillation->reseedTimer > 0) {
+            oscillation->reseedTimer--;
         } else {
-            temp_s0->unk_00 = Rand_S16Offset(30, 70);
-            temp_s0->unk_04 = Rand_S16Offset(D_80A676B8[i].unk_00, D_80A676B8[i].unk_02);
-            temp_s0->unk_0C = D_80A676B8[i].unk_04 + (D_80A676B8[i].unk_08 * Rand_ZeroOne());
-            temp_s0->unk_10 = fabsf(temp_s0->unk_0C - temp_s0->unk_08) * 0.033333335f;
+            oscillation->reseedTimer = Rand_S16Offset(30, 70);
+            oscillation->argRateOfChange =
+                Rand_S16Offset(sPrecessionDueToPlayerOscillationRandReseedParams[i].argRateOfChangeBase,
+                               sPrecessionDueToPlayerOscillationRandReseedParams[i].argRateOfChangeRange);
+            oscillation->maxMagnitude =
+                sPrecessionDueToPlayerOscillationRandReseedParams[i].maxMagnitudeBase +
+                (sPrecessionDueToPlayerOscillationRandReseedParams[i].maxMagnitudeRange * Rand_ZeroOne());
+            oscillation->magnitudeRateOfChange =
+                (1.0f / 30) * fabsf(oscillation->maxMagnitude - oscillation->currentMagnitude);
         }
-        temp_s0->unk_02 += temp_s0->unk_04;
-        Math_StepToF(&temp_s0->unk_08, temp_s0->unk_0C, temp_s0->unk_10);
-        temp_f22 += Math_SinS(temp_s0->unk_02) * temp_s0->unk_08;
+        oscillation->currentArg += oscillation->argRateOfChange;
+        Math_StepToF(&oscillation->currentMagnitude, oscillation->maxMagnitude, oscillation->magnitudeRateOfChange);
+        tiltMagnitude1 += Math_SinS(oscillation->currentArg) * oscillation->currentMagnitude;
     }
-
-    temp_f22 *= arg0->unk_08;
-
-    *arg2 = (s32)temp_f22 + arg0->unk_04;
+    tiltMagnitude1 *= dynamics->impactFromPlayer;
+    *tiltMagnitude = (s32)tiltMagnitude1 + dynamics->tiltMagnitude2;
 }
 
-void func_80A66C4C(ObjDrifticeStruct4* arg0, ObjDriftice* this, s16* arg2, s16* arg3) {
-    ObjDrifticeStruct3* temp_s0;
-    f32 temp_f20 = 0.0f;
+/*
+Updates the state of the dynamics associated with generic precession of platform (unrelated to player interaction).
+ Populates tiltMagnitude and precessionAngle
+
+Updates three oscillation states and sums their outputs into tiltMagnitude
+precessionAngle is the sum of two random angles which change over time. Always between -600 and 0.
```

Comment:
```suggestion
precessionAngle is the sum of two random angles which change over time. Always between -0x258 and 0.
```
We always try to have angles in hex